### PR TITLE
Icelandic language pack

### PR DESCRIPTION
**Pull request by lumpidu**
Adds Icelandic language pack:

 - Icelandic dictionary with appropriate frequencies
 - 1 keyboard layout
 - screen shots
 - Icelandic flag

No I8N for user texts yet: the English version was chosen.

There is a former Icelandic language pack, but its auto-completion is pretty poor, as there are no word frequencies provided. This PR provides a dictionary of ~219K Icelandic words with appropriate word frequencies. The keyboard layout is pretty classic. Some long-press keys are optimized for Icelandic usage.

We generated the word list from the Icelandic Gigaword Corpus: https://repository.clarin.is/repository/xmlui/handle/20.500.12537/192, created by [The Árni Magnússon Institue for Icelandic Studies](https://repository.clarin.is/repository/xmlui/browse?value=The%20%C3%81rni%20Magn%C3%BAsson%20Institue%20for%20Icelandic%20Studies&type=publisher). The corpus is CC-BY-4.0 Árni Magnússon Institute.

### CODE REVIEW

File: addons/languages/icelandic/apk/build.gradle [hunk 1]
```diff
@@ -0,0 +1,8 @@
+ext.shouldBePublished = true
+ext.override_app_id = 'com.anysoftkeyboard.languagepack.icelandic_oss'
```
Comment (menny) [inline addons/languages/icelandic/apk/build.gradle:2]: No need for that, you can remove it. We own that pack(com.anysoftkeyboard.languagepack.icelandic) but didn't migrate the code here.

File: addons/languages/icelandic/apk/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="com.anysoftkeyboard.languagepack.german">
```
Comment (menny) [inline addons/languages/icelandic/apk/src/main/AndroidManifest.xml:2]: Should be com.anysoftkeyboard.languagepack.icelandic

File: addons/languages/icelandic/pack/build.gradle [hunk 3]
```diff
@@ -0,0 +1,9 @@
+ext.status_icon_text = "is"
+
+ext.dictionaryInputPossibleCharacters = null//"".toCharArray()
+ext.dictionaryInputAdditionalInnerCharacters = null//"".toCharArray()
+ext.shouldGenerateDictionary = true
+ext.shouldBePublished = false
+ext.notPublishingReason = "Initial version has not been published yet"
```
Comment (menny) [inline addons/languages/icelandic/pack/build.gradle:7]: These should not be in the `pack`, and also, I think we can publish. Let's try without these

**Review by menny — CHANGES_REQUESTED** [review]
This is amazing!
Thank you.

I have a few requests, also please mention the source of the wordlist.

File: addons/languages/icelandic/apk/src/main/AndroidManifest.xml (hunk 2 above)
Comment (lumpidu) [reply addons/languages/icelandic/apk/src/main/AndroidManifest.xml:2]: right, c&p issue

File: addons/languages/icelandic/apk/build.gradle (hunk 1 above)
Comment (lumpidu) [reply addons/languages/icelandic/apk/build.gradle]: done

File: addons/languages/icelandic/apk/src/main/AndroidManifest.xml (hunk 2 above)
Comment (lumpidu) [reply addons/languages/icelandic/apk/src/main/AndroidManifest.xml]: done

File: addons/languages/icelandic/pack/build.gradle (hunk 3 above)
Comment (lumpidu) [reply addons/languages/icelandic/pack/build.gradle]: ok

**Comment by lumpidu** [comment]
Thx. for the quick review, I have changed the code parts and added the reference to the Dictionary file. I hope it was ok to resolve the quick fixes.

File: ime/app/src/main/res/values-is/strings.xml [hunk 4]
```diff
@@ -0,0 +1,958 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (menny) [inline ime/app/src/main/res/values-is/strings.xml:1]: This entire file will be overwritten from Crowdin: If you wish to help translate to Icelandic, please head to https://ask_translate.evendanan.net/project/anysoftkeyboard/invite

**Review by menny — APPROVED** [review]

**Comment by menny** [comment]
ah. addons/languages/icelandic/pack/src/main/res/drawable-mdpi/ic_status_icelandicn.png should be removed. Notice the extra `n` there

**Comment by menny** [comment]
Regarding the translations, see https://github.com/AnySoftKeyboard/AnySoftKeyboard/pull/3388
the file is empty now, since there are no translations for Icelandic in crowdin

File: ime/app/src/main/res/values-is/strings.xml (hunk 4 above)
Comment (lumpidu) [reply ime/app/src/main/res/values-is/strings.xml:1]: I'd put this into another PR. What do we do here ? Remove the file ? Btw: Firefox complains about the above URL

Comment (menny) [reply ime/app/src/main/res/values-is/strings.xml]: Yes, remove the file.
If you wish to donate Icelandic translation, use Crowdin: https://crwd.in/anysoftkeyboard (I've changed the URL, as you mentioned, there is a issue with the cert)

File: addons/languages/icelandic/pack/src/main/res/values-is/icelandic_pack_strings_dont_translate.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources xmlns:tools="http://schemas.android.com/tools"
+           xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2"
+           tools:ignore="MissingTranslation,TypographyDashes">
+
+    <string name="icelandic_dictionary">Íslenska</string>
+
+    <string name="is_keyboard">Íslenska</string>
+    <string name="is_keyboard_qwerty">Íslenska QWERTY</string>
```
Comment (menny) [inline addons/languages/icelandic/pack/src/main/res/values-is/icelandic_pack_strings_dont_translate.xml:9]: Seems to be unused: https://github.com/AnySoftKeyboard/AnySoftKeyboard/runs/6773630658?check_suite_focus=true#step:6:9281

**Review by menny — CHANGES_REQUESTED** [review]
sorry about this... One more change is needed.

File: ime/app/src/main/res/values-is/strings.xml (hunk 4 above)
Comment (lumpidu) [reply ime/app/src/main/res/values-is/strings.xml]: File removed. Thx for the hint on how the I8N works for this project.

File: addons/languages/icelandic/pack/src/main/res/values-is/icelandic_pack_strings_dont_translate.xml (hunk 5 above)
Comment (lumpidu) [reply addons/languages/icelandic/pack/src/main/res/values-is/icelandic_pack_strings_dont_translate.xml]: string ressource removed

**Comment by lumpidu** [comment]
Anything else that needs to be tackled ? The failed checks ?

**Comment by menny** [comment]
Merged. Do you want us to publish? Or are you doing that under your account?

**Comment by lumpidu** [comment]
Thx. You can publish it if you like.

**Pull request merged**